core-rust-bridge/src/main/java/com/radixdlt/statecomputer/commit/CommitRequest.java

```diff
@@ -92,32 +88,27 @@ public boolean equals(Object o) {
     if (o == null || getClass() != o.getClass()) return false;
     CommitRequest that = (CommitRequest) o;
     return Objects.equals(transactions, that.transactions)
-        && Objects.equals(stateVersion, that.stateVersion)
-        && Objects.equals(stateHash, that.stateHash)
-        && Arrays.equals(proofBytes, that.proofBytes)
-        && Objects.equals(postCommitVertexStoreBytes, that.postCommitVertexStoreBytes);
+        && Objects.equals(proof, that.proof)
+        && Arrays.equals(
+            postCommitVertexStoreBytes.or((byte[]) null),
+            that.postCommitVertexStoreBytes.or((byte[]) null));
   }
 
   @Override
   public int hashCode() {
-    int result = Objects.hash(transactions, stateVersion, stateHash, postCommitVertexStoreBytes);
```

Comment:
_(note: this change is unrelated, but I believe it has always been a bug [hash based on `byte[]` identity])_